controller/backup_controller.go

```diff
@@ -577,15 +602,31 @@ func (bc *BackupController) hasMonitor(backupName string) *engineapi.BackupMonit
 	return bc.monitors[backupName]
 }
 
-func (bc *BackupController) enableBackupMonitor(backup *longhorn.Backup, volume *longhorn.Volume, backupTargetClient *engineapi.BackupTargetClient, biChecksum string, engineClient engineapi.EngineClient) (*engineapi.BackupMonitor, error) {
+func (bc *BackupController) enableBackupMonitor(backup *longhorn.Backup, volume *longhorn.Volume, backupTargetClient *engineapi.BackupTargetClient, biChecksum string, binClient *engineapi.EngineBinary) (*engineapi.BackupMonitor, error) {
 	monitor := bc.hasMonitor(backup.Name)
 	if monitor != nil {
 		return monitor, nil
 	}
 
 	bc.monitorLock.Lock()
 	defer bc.monitorLock.Unlock()
-	monitor, err := engineapi.NewBackupMonitor(bc.logger, backup, volume, backupTargetClient, biChecksum, engineClient, bc.enqueueBackupForMonitor)
+
+	engine, err := bc.ds.GetVolumeCurrentEngine(volume.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	im, err := bc.ds.GetInstanceManager(engine.Status.InstanceManagerName)
+	if err != nil {
+		return nil, err
+	}
+
+	proxy, err := bc.ProxyHandler.GetClient(im, binClient)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
The same, I feel these statements can be consolidated to a util method/function.

Reply:
Addressed.